js/src/common/helpers/icon.tsx

```diff
@@ -1,11 +1,13 @@
+import { Attributes, Vnode } from 'mithril';
+
 /**
  * The `icon` helper displays an icon.
  *
  * @param {String} fontClass The full icon class, prefix and the icon’s name.
- * @param {Object} attrs Any other attributes to apply.
- * @return {Object}
+ * @param {Attributes} attrs Any other attributes to apply.
+ * @return {Vnode}
  */
-export default function icon(fontClass, attrs = {}) {
+export default function icon(fontClass: string, attrs: Attributes = {}): Vnode {
```

Comment:
Do we want to use `ComponentAttrs` instead of Mithril's `Attributes`, or do we only want to do that for our custom components?

Reply:
This is not necessarily a component, and it doesn't really matter as ComponentAttrs are mainly used for the `this.attrs` type hinting. It doesn't apply here.